src/modules/70-pipeline/components/PipelineStudio/PipelineTemplateBuilder/TemplatePipelineSpecifications/TemplatePipelineSpecifications.tsx

```diff
@@ -0,0 +1,227 @@
+  const { showError } = useToaster()
+  const { getString } = useStrings()
+  const formikRef = React.useRef<FormikProps<unknown> | null>(null)
+  const formRefDom = React.useRef<HTMLElement | undefined>()
+  const [inputsTemplate, setInputsTemplate] = React.useState<PipelineInfoConfig>()
+  const [allValues, setAllValues] = React.useState<PipelineInfoConfig>()
+  const [initialValues, setInitialValues] = React.useState<PipelineInfoConfig>()
+  const [formikErrors, setFormikErrors] = React.useState<FormikErrors<PipelineInfoConfig>>()
+  const [showFormError, setShowFormError] = React.useState<boolean>()
+  const dummyPipeline = useRef(pipeline)
+
+  const onChange = React.useCallback(
+    debounce(async (values: PipelineInfoConfig): Promise<void> => {
+      await updatePipeline({ ...pipeline, ...values })
+    }, 300),
+    [pipeline, updatePipeline]
+  )
+
+  const {
+    data: pipelineResponse,
+    error: pipelineError,
+    refetch: refetchPipeline,
+    loading: pipelineLoading
+  } = useMutateAsGet(useGetYamlWithTemplateRefsResolved, {
+    queryParams: {
+      ...getScopeBasedProjectPathParams(queryParams, scope),
+      pipelineIdentifier: pipeline.identifier,
+      repoIdentifier: gitDetails.repoIdentifier,
+      branch: gitDetails.branch,
+      getDefaultFromOtherRepo: true
+    },
+    body: {
+      originalEntityYaml: yamlStringify(dummyPipeline.current)
+    }
+  })
+
+  const {
+    data: templateInputSetYaml,
+    error: templateInputSetError,
+    refetch: refetchTemplateInputSet,
+    loading: templateInputSetLoading
+  } = useGetTemplateInputSetYaml({
+    templateIdentifier: templateRef,
+    queryParams: {
+      ...getScopeBasedProjectPathParams(queryParams, scope),
+      versionLabel: defaultTo(pipeline.template?.versionLabel, ''),
+      repoIdentifier: gitDetails.repoIdentifier,
+      branch: gitDetails.branch,
+      getDefaultFromOtherRepo: true
+    }
+  })
+
+  React.useEffect(() => {
+    if (!templateInputSetLoading) {
+      try {
+        const templateInputs = parse(defaultTo(templateInputSetYaml?.data, ''))
+        setInputsTemplate(templateInputs)
+      } catch (error) {
+        showError(error.message, undefined, 'template.parse.inputSet.error')
+      }
+    }
+  }, [templateInputSetLoading, templateInputSetYaml?.data])
+
+  React.useEffect(() => {
+    if (pipelineResponse?.data?.mergedPipelineYaml) {
+      setAllValues(parse(pipelineResponse?.data?.mergedPipelineYaml))
+    }
+  }, [pipelineResponse?.data?.mergedPipelineYaml])
+
+  React.useEffect(() => {
+    try {
+      const mergedTemplateInputs = merge({}, inputsTemplate, pipeline.template?.templateInputs)
+      setTemplateInputs(pipeline, mergedTemplateInputs)
+      updatePipeline(pipeline)
+      setInitialValues(cloneDeep(pipeline))
+    } catch (error) {
+      showError(error.message, undefined, 'template.parse.inputSet.error')
+    }
+  }, [inputsTemplate])
```

Comment:
Codacy has a fix for the issue: [React Hook React.useEffect has missing dependencies: 'pipeline', 'showError', and 'updatePipeline'. Either include them or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232314)
```suggestion
  }, [inputsTemplate, pipeline, showError, updatePipeline])
```